From 20d3ab338c732b9c60fcb39159b32132fbb37772 Mon Sep 17 00:00:00 2001
From: Kshitij <kgupta@habana.ai>
Date: Mon, 17 Feb 2025 08:12:54 +0200
Subject: [PATCH] [SW-217478] g3: sec: soft reset isec config fix

This patch skips ISEC configuration for D1
HL_BLOCK_TYPE_PCIE block in soft reset security
configuration.

TODO: As D1 PCIE block is not used, ISEC configuration
for D1 HL_BLOCK_TYPE_PCIE block should be skip in
security init also by updating autogenerated ISEC header.

Change-Id: I98a2d1735432675471c77e4869d8334d4a7e9e7d
Signed-off-by: Kshitij <kgupta@habana.ai>
---

diff --git a/drivers/misc/hl_security/gaudi3/security.c b/drivers/misc/hl_security/gaudi3/security.c
index dec4d3c..392737a 100644
--- a/drivers/misc/hl_security/gaudi3/security.c
+++ b/drivers/misc/hl_security/gaudi3/security.c
@@ -311,6 +311,10 @@
 		for (major = 0; major < cfg->addr.major; major++) {
 			minor = 0;
 			do {
+				if ((rst_src == HL_RESET_SRC_SOFT_RST) &&
+				    (cfg->addr.block_type == HL_BLOCK_TYPE_PCIE) &&
+				    (major > 0))
+					continue;
 				base_addr = cfg->addr.base_addr + (major * cfg->addr.major_offset)
 					    + (minor * cfg->addr.minor_offset) + HL_ISEC_REG_OFFSET;
 
